docs/_posts/examples/3400-01-13-filtering-points.html

```diff
@@ -0,0 +1,195 @@
+        'marker-symbol': 'bicycle',
+        title: 'Seersucker Bike Ride and Social'
+    },
+    geometry: {
+        coordinates: [-77.052477, 38.943951],
+        type: 'Point'
+    }
+}, {
+    type: 'Feature',
+    properties: {
+        description: "Jazz-influenced hip hop artist <a href=\'http://www.muhsinah.com\' target=\'_blank\' title=\'Opens in a new window\'>Muhsinah</a> plays the <a href=\'http://www.blackcatdc.com\'>Black Cat</a> (1811 14th Street NW) tonight with <a href=\'http://www.exitclov.com\' target=\'_blank\' title=\'Opens in a new window\'>Exit Clov</a> and <a href=\'http://godsilla.bandcamp.com\' target=\'_blank\' title=\'Opens in a new window\'>Gods’illa</a>. 9:00 p.m. $12.",
+        'marker-symbol': 'music',
+        title: 'Muhsinah'
+    },
+    geometry: {
+        coordinates: [-77.031706, 38.914581],
+        type: 'Point'
+    }
+}, {
+    type: 'Feature',
+    properties: {
+        description: "The Arlington Players' production of Stephen Sondheim's  <a href=\'http://www.thearlingtonplayers.org/drupal-6.20/node/4661/show\' target=\'_blank\' title=\'Opens in a new window\'><em>A Little Night Music</em></a> comes to the Kogod Cradle at The Mead Center for American Theater (1101 6th Street SW) this weekend and next. 8:00 p.m. ",
+        'marker-symbol': 'music',
+        title: 'A Little Night Music'
+    },
+    geometry: {
+        coordinates: [-77.020945, 38.878241],
+        type: 'Point'
+    }
+}, {
+    type: 'Feature',
+    properties: {
+        description: '<a href=\'http://www.truckeroodc.com/www/\' target=\'_blank\'>Truckeroo</a> brings dozens of food trucks, live music, and games to half and M Street SE (across from Navy Yard Metro Station) today from 11:00 a.m. to 11:00 p.m.',
+        'marker-symbol': 'music',
+        title: 'Truckeroo'
+    },
+    geometry: {
+        coordinates: [-77.007481, 38.876516],
+        type: 'Point'
+    }
+}];
+
+var map = new mapboxgl.Map({
+    container: 'map',
+    style: 'mapbox://styles/mapbox/streets-v8',
+    center: [-77.0316, 38.8924],
+    zoom: 11
+});
+
+map.on('load', function() {
+    map.addSource('geojson', {
+        'type': 'geojson',
+        'data':  {
+            'type': 'FeatureCollection',
+            'features': features
+        }
+    });
+
+    // Add items to the filter-group container and
+    // Bind a filtering action to them.
+    var filterGroup = document.getElementById('filter-group');
+
+    // Derive menu list by `marker-symbol`.
+    var list = features.reduce(function(memo, feature) {
+        var symbol = feature.properties['marker-symbol'];
+
+        // Don't capture duplicates
+        var contains = memo.some(function(d) {
+            return d === symbol;
+        });
+
+        if (!contains) memo.push(symbol);
+        return memo;
+    }, []);
+
+    list.forEach(function(symbol) {
```

Comment:
Instead of reducing a temporary list, iterate over `features` here directly and use `map.getLayer('poi-' + symbol)` to check if the layer already exists.